consensus/src/network.rs

```diff
@@ -275,9 +283,13 @@ impl NetworkTask {
                 },
                 Event::NewPeer(metadata) => {
                     debug!(remote_peer = metadata.remote_peer_id, "Peer connected");
+                    self.connections
+                        .write()
+                        .insert(metadata.remote_peer_id, metadata.application_protocols);
                 }
                 Event::LostPeer(metadata) => {
                     debug!(remote_peer = metadata.remote_peer_id, "Peer disconnected");
+                    self.connections.write().remove(&metadata.remote_peer_id);
```

Comment:
This is fine for now.  I think as I've eluded to earlier, this should just be handled in the `NetworkSender` or some abstraction so that every application doesn't have to keep track of the connections and their protocols, but instead just ask `NetworkSender` about which protocols are supported.  Likely will be done at a different time with the network abstraction fixes.